src/opnsense/mvc/app/models/OPNsense/Syslog/Syslog.php

```diff
@@ -0,0 +1,350 @@
+        $item->Source = $source;
+        $item->Filter = $filter;
+        $item->ActionType = $type;
+        $item->Target = $target;
+        $item->Category = $category;
+
+        $this->Modified = true;
+        $this->saveIfModified();
+    }
+
+    /**
+     * Remove Syslog target.
+     * @param $source program name, null if no program name
+     * @param $filter comma-separated list of selectors facility.level (without spaces)
+     * @param $type type of action (file, pipe, remote, all)
+     * @param $target action target
+     * @throws \ModelException
+     */
+    public function delTarget($source, $filter, $type, $target)
+    {
+        $source = str_replace(' ', '', $source);
+        $filter = str_replace(' ', '', $filter);
+        $type = trim($type);
+        $target = trim($target);
+
+        foreach($this->LogTargets->Target->__items as $uuid => $item)
+        {
+            if($item->Source->__toString() == $source
+            && $item->Filter->__toString() == $filter
+            && $item->ActionType->__toString() == $type
+            && $item->Target->__toString() == $target)
+            {
+                $this->LogTargets->Target->del($uuid);
+                $this->Modified = true;
+                $this->saveIfModified();
+            }
+        }
+    }
+
+    /**
+     * Add or update Syslog category. For mapping in GUI.
+     * @param $name category name
+     * @param $description category description
+     * @throws \ModelException
+     */
+    public function setCategory($name, $description)
+    {
+        $name = trim($name);
+        $description = trim($description);
+
+        foreach($this->LogCategories->Category->__items as $uuid => $category)
+        {
+            if($category->Name->__toString() == $name)
+            {
+                if($category->Description->__toString() == $description)
+                {
+                    return;
+                }
+
+                $category->Description = $description;
+                $this->Modified = true;
+                $this->saveIfModified();
+                return;
+            }
+        }
+
+        $category = $this->LogCategories->Category->add();
+        $category->Name = $name;
+        $category->Description = $description;
+        $this->Modified = true;
+        $this->saveIfModified();
+    }
+
+    /**
+     * Add or update Syslog category. For mapping in GUI.
+     * @param $name category name
+     * @param $description category description
+     * @throws \ModelException
+     */
+    public function setLocalSocket($path)
+    {
+        $path = trim($path);
+
+        foreach($this->LocalSockets->Socket->__items as $uuid => $socket)
+        {
+            if($socket->Path->__toString() == $path)
+            {
+                return;
+            }
+        }
+
+        $socketdir = dirname($path);
+        if (!is_dir($socketdir))
+        {
+            mkdir($socketdir, 0777, true);
```

Comment:
I think socket dir creation have to be removed from here to custom service plugin code. Because this directory owner is unknown here at this point. Or I have to request directory owner from plugin...